tests/test_constant_threshold_detector.py

```diff
@@ -3,151 +3,164 @@
-        test_detector.train(data)
-        assert (
-            test_detector.config.training_metadata.strategy
-            == ConstantThresholdStrategy.QUARTILE
+    )
+    test_metric = mock_metric(data=[5, 4, 7, 9, 15, 1, 0])
+    test_detector = build_detector("constant-detector", detector_config)
+    data = test_metric.query()
+    test_detector.train(data)
+    assert (
+        test_detector.config.hyperparameters.strategy
+        == ConstantThresholdStrategy.QUARTILE
+    )
+    assert test_detector.config.params.thresholds.weak_upper_threshold == 16.25
+    assert test_detector.config.params.thresholds.strong_upper_threshold == 24.5
+    assert test_detector.config.params.thresholds.weak_lower_threshold == -5.75
+    assert test_detector.config.params.thresholds.strong_lower_threshold == -14
+
+def test_build_detector_with_invalid_strategy_raises_build_error(mock_metric):
+    detector_config = dict(
+        hyperparameters =dict(
+            strategy="invalid strategy", weak_multiplier=1.5, strong_multiplier=3.0
         )
-        assert test_detector.config.hyperparameters.weak_upper_threshold == 16.25
-        assert test_detector.config.hyperparameters.strong_upper_threshold == 24.5
-        assert test_detector.config.hyperparameters.weak_lower_threshold == -5.75
-        assert test_detector.config.hyperparameters.strong_lower_threshold == -14
-
-    # def test_create_detector_with_invalid_strategy_raises_build_error(
-    #     self, mock_metric
-    # ):
-    #     detector_config = dict(
-    #         strategy="invalid strategy", weak_multiplier=3, strong_multiplier=5
-    #     )
-    #     test_metric = mock_metric()
-    #     with pytest.raises(ValueError) as exception:
-    #         test_detector = ConstantThresholdDetector(test_metric, detector_config)
-    #     assert (
-    #         str(exception.value)
-    #         == "'invalid strategy' is not a valid constant_threshold_strategy"
-    #     )
-
-    # def test_train_detector_with_invalid_sample_size_raises_build_error(
-    #     self, mock_metric
-    # ):
-    #     detector_config = dict(strategy="sigma", weak_multiplier=3, strong_multiplier=5)
-    #     test_metric = mock_metric(data=[3])
-    #     test_detector = ConstantThresholdDetector(test_metric, detector_config)
-    #     data = test_metric.query()
-    #     with pytest.raises(exceptions.DetectorBuilderError) as exception:
-    #         test_detector.train(data)
-    #     assert str(exception.value) == "Sample must have at least two elements"
+    )
+    with pytest.raises(ValueError) as exception:
+        build_detector("constant-detector", detector_config)
+    assert "'invalid strategy' is not a valid ConstantThresholdStrategy" in str(exception.value)
+
+
+@responses.activate
+def test_train_detector_with_invalid_sample_size_raises_build_error(mock_metric):
+    responses.add(responses.POST, "http://modelservice/modelservice/api/detectorMappings/findMatchingByTags",
```

Comment:
Out of curiosity, why modelservice/modelservice instead of just modelservice?

Reply:
It might be better to name the config adaptive alerting service.  The second one is part of the path, not the endpoint.  Here's how its formatted:`f"{self._url}/modelservice/api/detectorMappings/findMatchingByTags"`